Bottom: 658798d33cb5f6284644dec345a9d5e3352594f5
Top:    7f4f89881a0bede09b55de85a9391a1f7a504740
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-10-16 16:08:32 +0100

Add debug catch to triangle handling

If for any reason we are about to run off the end of the
triangle array, deliberately segfault so gdb can catch where.


---

diff --git a/src/hid/common/hidgl.h b/src/hid/common/hidgl.h
index b956a65..d9afb3d 100644
--- a/src/hid/common/hidgl.h
+++ b/src/hid/common/hidgl.h
@@ -42,6 +42,10 @@ hidgl_add_triangle (triangle_buffer *buffer,
                     GLfloat x2, GLfloat y2,
                     GLfloat x3, GLfloat y3)
 {
+  if (buffer->triangle_count == TRIANGLE_ARRAY_SIZE) {
+    fprintf (stderr, "hidgl_add_triangle: Attempt to add past end of array. Will now segfault.\n");
+    *(char *)0 = 0;
+  }
   buffer->triangle_array [buffer->coord_comp_count++] = x1;
   buffer->triangle_array [buffer->coord_comp_count++] = y1;
   buffer->triangle_array [buffer->coord_comp_count++] = x2;
